pennylane/drawer/tape_text.py

```diff
@@ -14,18 +14,46 @@
 """
 This module contains logic for the text based circuit drawer through the ``tape_text`` function.
 """
+# TODO: Fix the latter two pylint warnings
+# pylint: disable=too-many-arguments, too-many-branches, too-many-statements
 
+from dataclasses import dataclass
+from typing import Optional
 import pennylane as qml
-from pennylane.measurements import Expectation, Probability, Sample, Variance, State
+from pennylane.measurements import Expectation, Probability, Sample, Variance, State, MidMeasureMP
 
 from .drawable_layers import drawable_layers
-from .utils import convert_wire_order, unwrap_controls
+from .utils import convert_wire_order, unwrap_controls, find_mid_measure_cond_connections
 
 
-def _add_grouping_symbols(op, layer_str, wire_map):
+@dataclass
+class _Config:
+    """Dataclass containing attributes needed for updating the strings to be drawn for each layer"""
+
+    wire_map: dict
+    """Map between wire labels and their place in order"""
+
+    bit_map: dict
+    """Map between mid-circuit measurements and their corresponding bit in order"""
+
+    cur_layer: Optional[int] = None
+    """Current layer index that is being updated"""
```

Comment:
I might potentially recommend separating this property out.  All the other properties will be the same over the course of the entire call, while `cur_layer` will be constantly changing.

Reply:
I used a `dataclass` instead of a `namedtuple` just to fit that attribute inside without having to create a new `namedtuple` in every iteration, so your suggestion makes sense. Although I do like hiding `cur_layer` away because it's only used by one function